meerk40t/gui/panes/navigationpanels.py

```diff
@@ -933,104 +1044,154 @@ def matrix_updated(self):
+        dx = 0
+        dy = -self.context.navigate_jog
+        self.translateit(dx, dy)
+
+    def on_translate_up_10(self, event=None):  # wxGlade: Navigation.<event_handler>
+        dx = 0
+        dy = -self.context.navigate_jog * 10
+        self.translateit(dx, dy)
+
+    def on_translate_left_1(self, event=None):  # wxGlade: Navigation.<event_handler>
         dx = -self.context.navigate_jog
         dy = 0
-        self.context("translate %f %f\n" % (dx, dy))
-        self.matrix_updated()
+        self.translateit(dx, dy)
+
+    def on_translate_left_10(self, event=None):  # wxGlade: Navigation.<event_handler>
+        dx = -self.context.navigate_jog * 10
+        dy = 0
+        self.translateit(dx, dy)
 
-    def on_translate_right(self, event=None):  # wxGlade: Navigation.<event_handler>
+    def on_translate_right_1(self, event=None):  # wxGlade: Navigation.<event_handler>
         dx = self.context.navigate_jog
         dy = 0
-        self.context("translate %f %f\n" % (dx, dy))
-        self.matrix_updated()
+        self.translateit(dx, dy)
 
-    def on_translate_down(self, event=None):  # wxGlade: Navigation.<event_handler>
+    def on_translate_right_10(self, event=None):  # wxGlade: Navigation.<event_handler>
+        dx = self.context.navigate_jog * 10
+        dy = 0
+        self.translateit(dx, dy)
+
+    def on_translate_down_1(self, event=None):  # wxGlade: Navigation.<event_handler>
         dx = 0
         dy = self.context.navigate_jog
-        self.context("translate %f %f\n" % (dx, dy))
-        self.matrix_updated()
+        self.translateit(dx, dy)
+
+    def on_translate_down_10(self, event=None):  # wxGlade: Navigation.<event_handler>
+        dx = 0
+        dy = self.context.navigate_jog * 10
+        self.translateit(dx, dy)
 
     def on_reset(self, event=None):  # wxGlade: Navigation.<event_handler>
         self.context("reset\n")
         self.matrix_updated()
 
-    def on_rotate_ccw(self, event=None):  # wxGlade: Navigation.<event_handler>
-        spooler, input_driver, output = self.context.registered[
-            "device/%s" % self.context.root.active
-        ]
-        x = input_driver.current_x if input_driver is not None else 0
-        y = input_driver.current_y if input_driver is not None else 0
-        self.context("rotate %fdeg %f %f\n" % (-5, x, y))
-        self.matrix_updated()
+    def on_rotate_ccw_5(self, event=None):  # wxGlade: Navigation.<event_handler>
+        angle = -5.0
+        self.rotateit(angle)
+
+    def on_rotate_cw_5(self, event=None):  # wxGlade: Navigation.<event_handler>
+        angle = 5.0
+        self.rotateit(angle)
+
+    def on_rotate_ccw_90(self, event=None):  # wxGlade: Navigation.<event_handler>
+        angle = -90.0
+        self.rotateit(angle)
+
+    def on_rotate_cw_90(self, event=None):  # wxGlade: Navigation.<event_handler>
+        angle = 90.0
+        self.rotateit(angle)
+
+    @staticmethod
+    def skewed_value(stxt):
```

Comment:
This code could just be replaced with `Angle.parse(stext)` but other than that this whole thing is fine. And I'd be happy to merge it. What errors exist did so before. This is certainly a net gain.

Reply:
Well, actually given you accepted the whole meerkitten invite. I'd be happy to see it merged, you could merge it yourself.